infra/cache/kCouchbaseCacheWrapper.php

```diff
@@ -548,16 +553,29 @@ protected function doIncrement($key, $delta = 1)
 	/* (non-PHPdoc)
 	 * @see kBaseCacheWrapper::doSet()
 	 */
-	protected function doSet($key, $var, $expiry = 0)
+	protected function doSet($key, $var, $expiry = 0, $attempts = 2)
 	{
-		if($this->debug)
+		if ($this->debug)
 			KalturaLog::debug("Bucket name [$this->name] key [$key] var [" . print_r($var, true) . "]");
-			
-		$meta = $this->bucket->upsert($key, $var, array(
-			'expiry' => $expiry
-		));
-		
-		return is_null($meta->error);
+
+		while ($attempts > 0)
+		{
+			try
+			{
+				$meta = $this->bucket->upsert($key, $var, array(
+					'expiry' => $expiry
+				));
+
+				return is_null($meta->error);
+			} catch (CouchbaseException $e)
+			{
+				KalturaLog::err($e);
+			}
+			$attempts--;
+		}
+
+		KalturaLog::debug("No retries left for Couchbase upsert operation.");
+		return false;
```

Comment:
Until today , upsert has thrown unhandled exception ,  we need to keep this behavior here.